xdem/vcrs.py

```diff
@@ -0,0 +1,256 @@
+    ):
+        return "Ellipsoid"
+
+    # Define CRS in case EPSG or CRS was passed
+    if isinstance(vcrs_input, (int, CRS)):
+        if isinstance(vcrs_input, int):
+            vcrs = CRS.from_epsg(vcrs_input)
+        else:
+            vcrs = vcrs_input
+
+        # Raise errors if the CRS constructed is not vertical or has other components
+        if isinstance(vcrs, CRS) and not vcrs.is_vertical:
+            raise ValueError(
+                "New vertical CRS must have a vertical axis, '{}' does not "
+                "(check with `CRS.is_vertical`).".format(vcrs.name)
+            )
+        elif isinstance(vcrs, CRS) and (vcrs.is_vertical and len(vcrs.sub_crs_list) > 1):
+            warnings.warn(
+                "New vertical CRS has a vertical dimension but also other components, "
+                "extracting the first vertical reference only."
+            )
+            vcrs = [subcrs for subcrs in vcrs.sub_crs_list if subcrs.is_vertical][0]
+
+    # If a string was passed
+    else:
+        # If a name is passed, define CRS based on dict
+        if isinstance(vcrs_input, str) and vcrs_input.upper() in _vcrs_meta.keys():
+            vcrs_meta = _vcrs_meta[vcrs_input]
+            vcrs = CRS.from_epsg(vcrs_meta["epsg"])
+        # Otherwise, attempt to read a grid from the string
+        else:
+            if isinstance(vcrs_input, pathlib.Path):
+                grid = vcrs_input.name
+            else:
+                grid = vcrs_input
+            vcrs = _build_vcrs_from_grid(grid=grid)
+
+    return vcrs
+
+
+def _grid_from_user_input(vcrs_input: str | pathlib.Path | int | CRS) -> str | None:
+
+    # If a grid or name was passed, get grid name
+    if isinstance(vcrs_input, (str, pathlib.Path)):
+        # If the string is within the supported names
+        if isinstance(vcrs_input, str) and vcrs_input in _vcrs_meta.keys():
+            grid = _vcrs_meta[vcrs_input]["grid"]
+        # If it's a pathlib path
+        elif isinstance(vcrs_input, pathlib.Path):
+            grid = vcrs_input.name
+        # Or an ellipsoid
+        elif vcrs_input.lower() == "ellipsoid":
+            grid = None
+        # Or a string path
+        else:
+            grid = vcrs_input
+    # Otherwise, there is none
+    else:
+        grid = None
+
+    return grid
+
+
+def _transform_zz(
+    crs_from: CRS, crs_to: CRS, xx: NDArrayf, yy: NDArrayf, zz: MArrayf | NDArrayf | int | float
+) -> MArrayf | NDArrayf | int | float:
+    """
+    Transform elevation to a new 3D CRS.
+
+    :param crs_from: Source CRS.
+    :param crs_to: Destination CRS.
+    :param xx: X coordinates.
+    :param yy: Y coordinates.
+    :param zz: Z coordinates.
+
+    :return: Transformed Z coordinates.
+    """
+
+    # Transform the grid
+    transformer = Transformer.from_crs(crs_from=crs_from, crs_to=crs_to, always_xy=True)
+
+    # Will preserve the mask of the masked-array since pyproj 3.4
```

Comment:
This is quite essential. Pyproj 3.4 is now a minimum version. Fine by me, but it should be specified in the requirements and environment files.

Reply:
I tried to find a snarky comment saying wHaT abOUT mASKS! But instead I found this hehe

Reply:
Done!